torch/distributed/_tensor/random.py

```diff
@@ -99,210 +59,242 @@ def manual_seed(seed: int, device_mesh: DeviceMesh) -> None:
                 f"calling manual_seed function over {device_mesh} but received different seed values on ranks:",
                 f"seed on rank {dist.get_rank()} is {seed}, and seed on rank {rank} is {object}!",
             )
+    # instantiate a RNG tracker if haven't. By default DTensor uses an
+    # OffsetBasedRNGTracker to perform random operators.
+    global _rng_tracker
+    if not _rng_tracker:
+        _rng_tracker = OffsetBasedRNGTracker()
 
     # the current rank is in mesh
     if device_mesh.get_coordinate() is not None:
-        device_handle = _get_device_handle(device_mesh.device_type)
-        if device_handle:
-            device_handle.manual_seed(seed)
+        if isinstance(_rng_tracker, OffsetBasedRNGTracker):
+            _rng_tracker._manual_seed(seed)
         else:
-            raise NotImplementedError(
-                f"DTensor randomness only supports cuda/cuda-like device type, but got {device_mesh.device_type}"
-            )
+            raise RuntimeError("Unknown parallel RNG style!")
 
 
-def set_pre_op_offset(spec: DTensorSpec) -> None:
-    """Set the starting RNG offset for current device's local shard before actual
-    op execution. The pre_op_offset value should start from the current RNG offset
-    and increment by the size of local shard until it reaches the size of the whole
-    DTensor. For different ranks that hold the same DTensor shard, their pre_op_offset
-    will be the same.
+class CudaRNGStateTracker:
+    # we assume that the caller of all the methods would already
+    # check if the current device is a CUDA device
+    def __init__(self):
+        if not torch.cuda.is_available():
+            raise RuntimeError(f"{self.__class__.__name__} instantiation requires the presence of CUDA device")
 
-    Args:
-        spec (:class:`DTensorSpec`): the spec of the DTensor object on which
-            we prepare the offset for running random ops.
+        self._states = {}
+        self._devices = [torch.cuda.current_device()]
```

Comment:
The reason I store `current_device` into a list is, the `fork_rng()` takes in a `List` argument. Then it seems more appropriate to call it `_devices` because it is a list even though it only has one element. WDYT?